usecase/pokemon.go

```diff
@@ -0,0 +1,53 @@
+package usecase
+
+import (
+	"log"
+
+	"github.com/eugenio-garcia/Go-Dispatch-Bootcamp/model"
+)
+
+type PokemonService interface {
+	GetAllPokemons() (model.Pokemons, error)
+	GetPokemonById(id int) (*model.Pokemon, error)
+	// GetPokemonByName(name string) (*model.Pokemon, error)
+}
+
+type PokemonUsecase struct {
+	service PokemonService
+}
+
+func New(s PokemonService) *PokemonUsecase {
```

Comment:
It is a best practice that when you instate from a package through the New function, the value defined for the return is the interface and not the struct. This way the struct can remain hidden and the only thing exposed is the interface and the functions that are implemented.

```
func New(s PokemonService) PokemonService {
```

Reply:
Im not sure if I understand, do you mean the PokemonUseCase interface in this example? the problem I see is that New is expecting an instance of a PokemonUsecase and not the PokemonService.